oak-lucene/src/main/java/org/apache/jackrabbit/oak/plugins/index/lucene/LuceneIndexProviderService.java

```diff
@@ -514,11 +525,13 @@ private void registerIndexEditor(BundleContext bundleContext, IndexTracker track
         if (enableCopyOnWrite){
             initializeIndexCopier(bundleContext, config);
             editorProvider = new LuceneIndexEditorProvider(indexCopier, tracker, extractedTextCache,
-                    augmentorFactory,  mountInfoProvider, activeDeletedBlobCollector, mBean, statisticsProvider);
+                    augmentorFactory,  mountInfoProvider, activeDeletedBlobCollector, mBean, statisticsProvider)
+                    .withAsyncIndexesSizeStatsUpdate(asyncIndexesSizeStatsUpdate);
             log.info("Enabling CopyOnWrite support. Index files would be copied under {}", indexDir.getAbsolutePath());
         } else {
             editorProvider = new LuceneIndexEditorProvider(null, tracker, extractedTextCache, augmentorFactory,
-                    mountInfoProvider, activeDeletedBlobCollector, mBean, statisticsProvider);
+                    mountInfoProvider, activeDeletedBlobCollector, mBean, statisticsProvider)
+                    .withAsyncIndexesSizeStatsUpdate(asyncIndexesSizeStatsUpdate);
         }
         editorProvider.setBlobStore(blobStore);
```

Comment:
you could remove lines 529 and 534 and add the async stats update here

```suggestion
        editorProvider
            .withAsyncIndexesSizeStatsUpdate(asyncIndexesSizeStatsUpdate)
            .setBlobStore(blobStore);
```